### ISSUE TITLE
aws_cloudwatch_log_metric_filter pattern character limit incorrect

### ISSUE THREAD

**Issue by michaelmcguinness** [terraform-providers/terraform-provider-aws#1810]
### Terraform Version
0.10.7

### Affected Resource(s)
aws_cloudwatch_log_metric_filter

### Issue
The pattern argument is restricted to 512 characters. I believe this should be 1024 characters as per the documentation here: http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutMetricFilter.html

Manual configuration confirms that patterns can be greater than 512 characters.

### Terraform Code
```func resourceAwsCloudWatchLogMetricFilter() *schema.Resource {
	return &schema.Resource{
		Create: resourceAwsCloudWatchLogMetricFilterUpdate,
		Read:   resourceAwsCloudWatchLogMetricFilterRead,
		Update: resourceAwsCloudWatchLogMetricFilterUpdate,
		Delete: resourceAwsCloudWatchLogMetricFilterDelete,

		Schema: map[string]*schema.Schema{
			"name": &schema.Schema{
				Type:         schema.TypeString,
				Required:     true,
				ForceNew:     true,
				ValidateFunc: validateLogMetricFilterName,
			},

			"pattern": &schema.Schema{
				Type:         schema.TypeString,
				Required:     true,
				ValidateFunc: validateMaxLength(512),
				StateFunc: func(v interface{}) string {
					s, ok := v.(string)
					if !ok {
						return ""
					}
					return strings.TrimSpace(s)
				},
			},
```

**Comment by atsushi-ishibashi** [comment]
Fixed #2107

**Comment by Ninir** [comment]
Thanks for the ping @atsushi-ishibashi :)
